### DISCUSSION TITLE
Google Coral Edge TPU support?

### DISCUSSION THREAD

**Discussion by Alphakilo** [nextcloud/recognize#681] [Q&A] [answered]
Hi there 👋 

I wanted to ask whether there's the possibility to use a [Google Coral TPU](https://coral.ai/products/accelerator) to speed up classification?

Thanks!

**Comment by muebau** [comment]
I think it could be helpful to have some kind of interface like suggested in #680 . It should be easy to comply to such interface with some docker container.
As there is WebUSB support now (https://github.com/google-coral/webcoral) it should be easy to create a browser version like I suggested in #680 too.

**Reply by Ayo91** [reply]
amazing idea!

**Comment by Fujimarudelviento** [comment]
As Tensorflow is used by Nextcloud it should be fairly simple to replicate how other projects like Frigate NVR leverage Google Coral TPU, see here https://docs.frigate.video/frigate/hardware, at least that's what I in my infinite ignorance think :)

**Comment by marcelklehr** [comment, marked as answer]
> The Coral Edge TPU is capable of executing deep feed-forward neural networks such as convolutional neural networks (CNN). It supports only TensorFlow Lite models that are fully 8-bit quantized and then compiled specifically for the Edge TPU.

Recgonize does not exclusively use TF lite models. Anything that works in WASM mode probably also works in TF lite, but it would be an additional effort to compile models for TF lite and add support for coral tpus.

**Reply by Ayo91** [reply]
So, Coral TPUs are not supported in Nextcloud, which gpus do you recommend for speed up the recognition processes?

**Comment by vchrizz** [comment]
In case you missed it:
[The chip made for the AI inference era – the Google TPU](https://www.uncoveralpha.com/p/the-chip-made-for-the-ai-inference)
[Google TPUs Vs Nvidia GPUs](https://www.forbes.com/sites/greatspeculations/2025/09/11/google-tpus-vs-nvidia-gpus/)
[Nvidia Blackwell, Google TPUs, AWS Trainium: Comparing top AI chips](https://www.cnbc.com/2025/11/21/nvidia-gpus-google-tpus-aws-trainium-comparing-the-top-ai-chips.html)
[How Do Google’s TPU AI Chips Differ From Nvidia GPUs? - Bloomberg](https://www.bloomberg.com/news/articles/2025-11-25/how-do-google-s-tpu-ai-chips-differ-from-nvidia-gpus)
[Meta, Google discuss TPU deal as Google targets Nvidia’s lead, Information says](https://finance.yahoo.com/news/meta-google-discuss-tpu-deal-233823637.html)
